### PR TITLE
Remove support for Docker Compose v1

### PR DESCRIPTION
**Pull request by jsoriano**
elastic-package is likely not working with Docker Compose v1 at this point, and it is discontinued since last year. Completely remove support for it.

### CODE REVIEW

File: internal/compose/compose.go [hunk 1]
```diff
@@ -199,20 +198,13 @@ func NewProject(name string, paths ...string) (*Project, error) {
...
 	ver, err := c.dockerComposeVersion(context.Background())
 	if err != nil {
-		logger.Errorf("Unable to determine Docker Compose version: %v. Defaulting to 1.x", err)
-		c.dockerComposeV1 = true
+		logger.Errorf("Unable to determine Docker Compose version: %v", err)
 		return &c, nil
 	}
-
-	versionMessage := fmt.Sprintf("Determined Docker Compose version: %v", ver)
-	if ver.Major() == 1 {
```
Comment (mrodm) [inline internal/compose/compose.go:208]: Should we keep this check and raise an error if versions 1.* are used ?

File: internal/compose/compose.go [hunk 2]
```diff
@@ -199,20 +198,15 @@ func NewProject(name string, paths ...string) (*Project, error) {
 	// Passing a nil context here because we are on initialization.
 	ver, err := c.dockerComposeVersion(context.Background())
 	if err != nil {
-		logger.Errorf("Unable to determine Docker Compose version: %v. Defaulting to 1.x", err)
-		c.dockerComposeV1 = true
-		return &c, nil
+		return nil, fmt.Errorf("unable to determine Docker Compose version: %w", err)
```
Comment (jsoriano) [inline internal/compose/compose.go:201]: Making this an error now, the main reason for this to fail is that docker compose command is not found.

File: internal/compose/compose.go (hunk 1 above)
Comment (jsoriano) [reply internal/compose/compose.go:208]: Kept, good point.

**Review by mrodm — APPROVED** [review]
:rocket:

**Pull request merged**